azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -126,17 +129,33 @@ def create_from_edge_environment(cls):
             workload_uri = os.environ["IOTEDGE_WORKLOADURI"]
             api_version = os.environ["IOTEDGE_APIVERSION"]
         except KeyError:
-            raise auth.IoTEdgeError("IoT Edge container not configured correctly")
-
-        authentication_provider = auth.IoTEdgeAuthenticationProvider(
-            hostname=hostname,
-            device_id=device_id,
-            module_id=module_id,
-            gateway_hostname=gateway_hostname,
-            module_generation_id=module_generation_id,
-            workload_uri=workload_uri,
-            api_version=api_version,
-        )
+            # As a fallback, try the Edge local dev variables for debugging.
+            # These variables are set by VS/VS Code in order to allow debugging
+            # of Edge application code in a non-Edge dev environment.
+            try:
+                connection_string = os.environ["EdgeHubConnectionString"]
+                ca_cert = os.environ["EdgeModuleCACertificateFile"]
+            except KeyError:
+                # TODO: consider using a different error here.
```

Comment:
># TODO: consider using a different error here [](start = 16, length = 45)

worth checking in the guidelines: at least in Typescript they have guidance for what errors to fire for configuration issues. maybe the python guidelines have something similar.
Also: please create an item in the backlog rather than a TODO in the code :)

Reply:
Will resolve this as part of the error story revisions

---
In reply to: [297895687](https://github.com/Azure/azure-iot-sdk-python-preview/pull/117#discussion_r297895687) [](ancestors = 297895687)